File: 0792-binary-search/0792-binary-search.py
# Time: O(logn)
class Solution:
	def search(self, nums: List[int], target: int) -> int:
		def bSearch(nums, low, high, targ):
			if high >= low:
				mid = low+(high-low)//2
				if nums[mid] == targ:
					return mid
				elif nums[mid] > targ:
					return bSearch(nums, low, mid-1, targ)
				else:
					return bSearch(nums, mid+1, high, targ)
			else:
				return -1
		return bSearch(nums, 0, len(nums)-1, target)

File: 0792-binary-search/0792-binary-search.java
class Solution {
          public int help(int[] nums, int target, int l, int r){
            if (l <=r){
               int mid = l + (r - l) / 2;
               if (nums[mid] == target){
                  return mid;
               }
               if (nums[mid] < target){
                  return help(nums, target, mid + 1, r);
               }
               return help(nums, target, l, mid - 1);
            }
            return -1;
         }
    public int search(int[] nums, int target) {
        return help(nums, target, 0, nums.length - 1);
    }
}